ai21/clients/sagemaker/resources/sagemaker_completion.py

```diff
@@ -1,45 +1,43 @@
-from typing import Optional, List
+from typing import List
 
 from ai21.clients.sagemaker.resources.sagemaker_resource import SageMakerResource
 from ai21.models import Penalty, CompletionsResponse
+from ai21.types import NotGiven, NOT_GIVEN
+from ai21.utils.typing import remove_not_given
 
 
 class SageMakerCompletion(SageMakerResource):
     def create(
         self,
         prompt: str,
         *,
-        max_tokens: Optional[int] = None,
-        num_results: Optional[int] = 1,
-        min_tokens: Optional[int] = 0,
-        temperature: Optional[float] = 0.7,
-        top_p: Optional[int] = 1,
-        top_k_return: Optional[int] = 0,
-        stop_sequences: Optional[List[str]] = None,
-        frequency_penalty: Optional[Penalty] = None,
-        presence_penalty: Optional[Penalty] = None,
-        count_penalty: Optional[Penalty] = None,
+        max_tokens: int | NotGiven = NOT_GIVEN,
+        num_results: int | NotGiven = NOT_GIVEN,
+        min_tokens: int | NotGiven = NOT_GIVEN,
+        temperature: float | NotGiven = NOT_GIVEN,
+        top_p: float | NotGiven = NOT_GIVEN,
+        top_k_return: int | NotGiven = NOT_GIVEN,
+        stop_sequences: List[str] | NotGiven = NOT_GIVEN,
+        frequency_penalty: Penalty | NotGiven = NOT_GIVEN,
+        presence_penalty: Penalty | NotGiven = NOT_GIVEN,
+        count_penalty: Penalty | NotGiven = NOT_GIVEN,
         **kwargs,
     ) -> CompletionsResponse:
-        body = {
-            "prompt": prompt,
-            "maxTokens": max_tokens,
-            "numResults": num_results,
-            "minTokens": min_tokens,
-            "temperature": temperature,
-            "topP": top_p,
-            "topKReturn": top_k_return,
-            "stopSequences": stop_sequences or [],
-        }
-
-        if frequency_penalty is not None:
-            body["frequencyPenalty"] = frequency_penalty.to_dict()
-
-        if presence_penalty is not None:
-            body["presencePenalty"] = presence_penalty.to_dict()
-
-        if count_penalty is not None:
-            body["countPenalty"] = count_penalty.to_dict()
+        body = remove_not_given(
+            {
+                "prompt": prompt,
+                "maxTokens": max_tokens,
+                "numResults": num_results,
+                "minTokens": min_tokens,
+                "temperature": temperature,
+                "topP": top_p,
+                "topKReturn": top_k_return,
+                "stopSequences": stop_sequences or [],
+                "frequencyPenalty": frequency_penalty.to_dict() if frequency_penalty else frequency_penalty,
```

Comment:
I believe you need to check for `NOT_GIVEN` here as we no longer pass `None`

Reply:
`__bool__` is False so it gets checked